File: IAIso-v5.0/core/iaiso-php/src/Conformance/SectionResults.php
<?php

declare(strict_types=1);

namespace IAIso\Conformance;

/** Results aggregated by section. */
final class SectionResults
{
    /** @var VectorResult[] */
    public array $pressure = [];
    /** @var VectorResult[] */
    public array $consent = [];
    /** @var VectorResult[] */
    public array $events = [];
    /** @var VectorResult[] */
    public array $policy = [];

    public function countPassed(): int
    {
        $n = 0;
        foreach ([$this->pressure, $this->consent, $this->events, $this->policy] as $bucket) {
            foreach ($bucket as $r) if ($r->passed) $n++;
        }
        return $n;
    }

    public function countTotal(): int
    {
        return count($this->pressure) + count($this->consent)
            + count($this->events) + count($this->policy);
    }
}

File: IAIso-v5.0/core/iaiso-java/iaiso-conformance/src/main/java/io/iaiso/conformance/SectionResults.java
package io.iaiso.conformance;

import java.util.ArrayList;
import java.util.List;

/** Results aggregated by section. */
public final class SectionResults {
    public final List<VectorResult> pressure = new ArrayList<>();
    public final List<VectorResult> consent = new ArrayList<>();
    public final List<VectorResult> events = new ArrayList<>();
    public final List<VectorResult> policy = new ArrayList<>();

    public int countPassed() {
        int p = 0;
        for (List<VectorResult> bucket : new List[]{pressure, consent, events, policy}) {
            for (VectorResult r : bucket) if (r.passed) p++;
        }
        return p;
    }

    public int countTotal() {
        return pressure.size() + consent.size() + events.size() + policy.size();
    }
}
